sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltinFun.scala

```diff
@@ -1266,13 +1269,27 @@ private[lf] object SBuiltinFun {
   // matches the template type, and then return the SAny internal value.
   final case class SBFromInterface(
       tplId: TypeConName
-  ) extends SBuiltinPure(1) {
-    override private[speedy] def executePure(args: util.ArrayList[SValue]): SOptional = {
+  ) extends SBuiltinFun(1) {
+    override private[speedy] def execute[Q](
+        args: util.ArrayList[SValue],
+        machine: Machine[Q],
+    ): Control[Q] = {
       val (tyCon, record) = getSAnyContract(args, 0)
+
       if (tplId == tyCon) {
-        SOptional(Some(record))
+        Control.Value(SOptional(Some(record)))
+      } else if (tplId.qualifiedName == tyCon.qualifiedName) {
+        val (tplIdPkgName, _) = machine.tmplId2PackageNameVersion(tplId)
+        val (tyConPkgName, _) = machine.tmplId2PackageNameVersion(tyCon)
+        if (tplIdPkgName == tyConPkgName) {
+          importValue(machine, tplId, record.toUnnormalizedValue) { templateArg =>
+            Control.Value(SOptional(Some(templateArg)))
+          }
```

Comment:
This isn't good enough for 3x, but its a larger task to fix. For now, I'm just aligning the branches

Reply:
Maybe leave a note about why this is not good enough.